tests/SideBySide.New/AppConfig.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+
+using MySql.Data.MySqlClient;
+
+namespace SideBySide
+{
+    public static class AppConfig
+    {
+	    private static IReadOnlyDictionary<string, string> DefaultConfig{get;} =
+		    new Dictionary<string, string>
+		    {
+			    ["Data:NoPasswordUser"] = "",
+			    ["Data:SupportsJson"] = "false",
+		    };
+
+	    private static int _configFirst;
+	    private static IConfiguration ConfigBuilder { get; } = new ConfigurationBuilder()
+		    .SetBasePath(Path.GetFileName(Directory.GetCurrentDirectory()) == "SideBySide.New" ? Directory.GetCurrentDirectory() :
+		    	Path.Combine(Directory.GetCurrentDirectory(), "tests", "SideBySide.New"))
+	    	.AddInMemoryCollection(DefaultConfig)
+		    .AddJsonFile("config.json")
+		    .Build();
+
+	    public static IConfiguration Config
+	    {
+		    get
+		    {
+			    if (Interlocked.Exchange(ref _configFirst, 1) == 0)
+				    Console.WriteLine("Config Read");
```

Comment:
I was planning to use this from CI down the road.  If we are going to test 8 different combinations per database, we will want to do that in parallel with 8 different processes.  The "launching script" would read STDOUT until it sees "Config Read", then it would know it can swap the config.json out and launch the next process.